packages/asc-ui/src/components/Typography/TypographyStyle.ts

```diff
@@ -24,11 +25,16 @@ export const defaultTypographyStyles = {
 
 export type Variant = keyof typeof defaultTypographyStyles
 
-const getProperty = <T, K extends keyof T>(obj: T, key: K) => obj[key]
-
 export default styled.p<Props>`
-  ${({ as, forwardedAs }) =>
-    getProperty(defaultTypographyStyles, as || forwardedAs)};
+  ${({ as, forwardedAs }) => {
+    const key = as ?? forwardedAs
+
+    if (typeof key === 'string') {
```

Comment:
Does this work?
`return typeof key === 'string' && defaultTypographyStyles[key] ?? null`?

Reply:
Actually, the function can return any falsy value so we can even drop the `null`. Changed it.